pydrad/parse/parse.py

```diff
@@ -525,6 +436,102 @@ def grid_edges_right(self) -> u.cm:
         """
         return self.grid_edges[1:]
 
+    @property
+    @u.quantity_input
+    def coordinate(self) -> u.cm:
+        """
+        Coordinate in the field-aligned direction
+        """
+        return self.grid_centers
+
+    @property
+    @u.quantity_input
+    def electron_mass_density(self) -> u.Unit('g cm-3'):
+        # TODO: Account for possible presence of both electron
+        # and ion mass densities. If the electron mass density
+        # is present in this file, it will shift all of the
+        # indices in the .amr file by one.
+        if 'electron_mass_density' in self._amr_data.colnames:
+            return self._amr_data['electron_mass_density']
+        return self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_mass_density(self) -> u.Unit('g cm-3'):
+        return self._amr_data['ion_mass_density']
+
+    @property
+    @u.quantity_input
+    def momentum_density(self) -> u.Unit('g cm-2 s-1'):
+        return self._amr_data['momentum_density']
+
+    @property
+    @u.quantity_input
+    def electron_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['electron_energy_density']
+
+    @property
+    @u.quantity_input
+    def ion_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['ion_energy_density']
+
+    @property
+    @u.quantity_input
+    def velocity(self) -> u.cm/u.s:
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['velocity']
+        return self.momentum_density / self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_density(self) -> u.Unit('cm-3'):
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['ion_density']
+        # NOTE: This is pulled directly from the HYDRAD source code
+        # https://github.com/rice-solar-physics/HYDRAD/blob/master/Resources/source/constants.h
+        # and is the average ion mass for a H-He plasma computed by weighting the H and He ion
+        # masses with a particular set of abundances.
+        m_ion = 2.171e-24 * u.g
+        return self.ion_mass_density / m_ion
+
+    @property
+    @u.quantity_input
+    def electron_density(self) -> u.Unit('cm-3'):
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['electron_density']
+        # FIXME: If this exists as a separate column in the .amr file then
+        return self.ion_density
```

Comment:
I think that the default value depends on which switches are turned on in HYDRAD.  With OPTICALLY_THICK_RADIATION on, the two densities are set to a small factor apart:
https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L1181

Reply:
Ah ok. Currently, the parsing of the results doesn't actually depend at all on the configuration so it's not quite as simple as just checking whether this option is set. I did this explicitly so you could still parse runs which weren't created with `pydrad`. There could be a kwarg that you pass in explicitly that deals with this at the file parsing level. Alternatively, we could always *try* to look for configuration and adjust accordingly and if it can't be found just fall back to the default behavior.

Reply:
This one is really complicated.  If neither `OPTICALLY_THICK_RADIATION` nor `NLTE_CHROMOSPHERE` is on, the two densities are equal (I think).  If OTR is on, and NLTE off, they're set a small factor apart.  If both are on, the electron density is calculated by summing over the contributions from all elements, which also then depends on the assumed abundance set.  

This is certainly the simplest way to deal with it, because I could imagine the calculation is otherwise rather difficult to recreate.  Maybe keep this, and create a new issue to generalize this with a kwarg switch?